src/framework/preferences/js/PrefsEditor.js

```diff
@@ -391,13 +391,17 @@ var fluid_2_0 = fluid_2_0 || {};
             fluid.model.diff(userSelections.preferences, fluid.get(that.initialModel, ["preferences"]), stats);
 
             fluid.each(stats.changeMap, function (state, path) {
-                fluid.set(changedSelections, ["preferences", path], userSelections.preferences[path]);
+                fluid.set(changedPrefs, path, userSelections.preferences[path]);
             });
+
+            if (stats.changes > 0) {
+                userSelections.preferences = changedPrefs;
+            }
         }
 
-        that.events.onSave.fire(changedSelections);
-        that.setSettings(changedSelections);
-        return changedSelections;
+        that.events.onSave.fire(userSelections);
+        that.setSettings(userSelections);
+        return userSelections;
```

Comment:
To inform what has been saved. The saved pieces are changed prefs and other parts in the model.